docs/porting/custom-target-porting.md

```diff
@@ -0,0 +1,163 @@
+
+ImaginaryBoard is the same as the standard NRF52840 development kit, but you want to remove some unused features, such as AnalogIn, and change the way some pins are connected.
+
+1. Choose a target in Mbed OS that your board can inherit from. 
+
+   If that existing target follows one of the recommended inheritance patterns, such as Family -> MCU -> Board, then it makes it simple to just inherit the MCU target information and add your own board information. In some cases, you may need to inherit a Family or Board and add or replace configurations accordingly.
+
+   In this example, there is an MCU target configuration in `mbed-os/targets/targets.json` that the new board can inherit from. It is called MCU_NRF52840.
+   
+   Below is a short summary of the MCU target. For simplicity, areas have been omitted.
+   
+   From mbed-os/targets/targets.json:
+   
+   ```
+       "MCU_NRF52840": {
+           "inherits": ["Target"],
+           "components_add": […],
+           "core": "Cortex-M4F",
+           "macros": […],
+           "features": […],
+           "device_has": […],
+           "extra_labels": […],
+           "config": {…},
+           "overrides": {…},
+           "OUTPUT_EXT": "hex",
+           "supported_toolchains": ["GCC_ARM", "ARM", "IAR"],
+           "public": false,
+           "detect_code": ["1101"],
+           "bootloader_supported": true
+       }
+   ```
+   
+   For reference, the standard NRF52840 development kit configuration is below. It inherits from the MCU target.
+   
+   ```
+       "NRF52840_DK": {
+           "supported_form_factors": ["ARDUINO"],
+           "inherits": ["MCU_NRF52840"],
+           "release_versions": ["5"],
+           "device_name": "nRF52840_xxAA"
+       },
+   ```
+   
+   In this example, the new board is similar to the NRF52840 development kit. 
+   
+1. Instead of making edits within `mbed-os`, create a `custom_target.json` file, and put it in a directory alongside `mbed-os`.
+   
+   For example:  
+   
+   ```
+   {
+   "IMAGINARYTARGET": {
+           "inherits": ["MCU_NRF52840"],
+           "release_versions": ["5"],
+           "device_name": "nRF52840_xxAA",
+           "detect_code": ["1234"],
+           "device_has_remove": ["ANALOGIN"],
+           “components_remove”: ["QSPI"]
+       }
+   }
+   ```
+   
+   The new board target configuration inherits the MCU target, then follows similar configurations to the development kit, including specifying it should build with Mbed OS 5. Also, it uses `device_name nRF52840_xxAA`, which matches the CMSIS pack for that device. The `detect_code` is a unique ID that identifies the board to the Mbed OS test tools.
+   
+   <span class="notes">**Note:** Contact the Arm Mbed team to get a unique ID, especially if you plan to have your board coexist with other Mbed Enabled boards while running automated tests.</span>
```

Comment:
Oh, is that the detect_code? Then maybe break it out of line 72 so that it more clearly goes with a comment.